packages/dds/tree/src/test/simple-tree/simpleTree.bench.ts

```diff
@@ -0,0 +1,128 @@
+	type WideTreeNode,
+} from "./utils.js";
+
+// number of nodes in test for wide trees
+const nodesCountWide = [
+	[1, BenchmarkType.Measurement],
+	[100, BenchmarkType.Perspective],
+	[500, BenchmarkType.Measurement],
+];
+// number of nodes in test for deep trees
+const nodesCountDeep = [
+	[1, BenchmarkType.Measurement],
+	[10, BenchmarkType.Perspective],
+	[100, BenchmarkType.Measurement],
+];
+
+describe("SimpleTree benchmarks", () => {
+	describe("Read SimpleTree", () => {
+		const leafValue = 1;
+		for (const [numberOfNodes, benchmarkType] of nodesCountDeep) {
+			let tree: DeepTreeNode;
+			let actualDepth = 0;
+			let actualValue = 0;
+
+			benchmark({
+				type: benchmarkType,
+				title: `Deep Tree as SimpleTree: reads with ${numberOfNodes} nodes`,
+				before: () => {
+					// Setup
+					tree = generateDeepSimpleTree(numberOfNodes, leafValue);
+				},
+				benchmarkFn: () => {
+					const { depth, value } = readDeepSimpleTree(tree);
+					actualDepth = depth;
+					actualValue = value;
+				},
+				after() {
+					//  Assert read values
+					assert.equal(actualDepth, numberOfNodes);
+					assert.equal(actualValue, leafValue);
+				},
+			});
+		}
+		for (const [numberOfNodes, benchmarkType] of nodesCountWide) {
+			let tree: WideTreeNode;
+			let expected = 0;
+			let actualNodesCount = 0;
+			let actualSum = 0;
+			benchmark({
+				type: benchmarkType,
+				title: `Wide Tree as SimpleTree: reads with ${numberOfNodes} nodes`,
+				before: () => {
+					// Setup
+					tree = generateWideSimpleTree(numberOfNodes, leafValue);
+					for (let i = 0; i < numberOfNodes; i++) {
+						expected += leafValue;
+					}
+				},
+				benchmarkFn: () => {
+					const { nodesCount, sum } = readWideSimpleTree(tree);
+					actualNodesCount = nodesCount;
+					actualSum = sum;
+				},
+				after() {
+					assert.equal(actualNodesCount, numberOfNodes);
+					assert.equal(actualSum, expected);
+				},
+			});
+		}
+	});
+
+	describe(`Edit SimpleTree`, () => {
+		const leafValue = 1;
+		const changedLeafValue = -1;
+		for (const [numberOfNodes, benchmarkType] of nodesCountDeep) {
+			let tree: DeepTreeNode;
+			benchmark({
+				type: benchmarkType,
+				title: `Update value at leaf of ${numberOfNodes} deep tree`,
+				before: () => {
+					// Setup
+					tree = generateDeepSimpleTree(numberOfNodes, leafValue);
+				},
+				benchmarkFn: () => {
+					writeDeepSimpleTree(tree, changedLeafValue);
+				},
+				after: () => {
+					const expected = generateDeepSimpleTree(numberOfNodes, changedLeafValue);
+					assert.deepEqual(tree, expected);
+				},
+			});
+		}
+		for (const [numberOfNodes, benchmarkType] of nodesCountWide) {
+			let tree: WideTreeNode;
+			benchmark({
+				type: benchmarkType,
+				title: `Update value at leaf of ${numberOfNodes} Wide tree`,
+				before: () => {
+					// Setup
+					tree = generateWideSimpleTree(numberOfNodes, leafValue);
+				},
+				benchmarkFn: () => {
+					writeWideSimpleTree(tree, changedLeafValue);
```

Comment:
Let's use this version of the edit to begin with.
```suggestion
					writeWideSimpleTreeEndValue(tree, changedLeafValue);
```